src/runtime/pythonengine.cs

```diff
@@ -538,6 +546,56 @@ public static GILState GIL()
 
             return new GILState();
         }
+        
+        private static Dictionary<string, PyScope> NamedScopes = new Dictionary<string, PyScope>();
```

Comment:
I don't think that this is the right place for the global scope storage. Is there any particular reason for not putting this into the `PyScope` class?